File: 1544. Make The String Great.py
class Solution:
    def makeGood(self, s: str) -> str:
        cur = 0
        
        while cur<len(s)-1:
            if cur>=0 and ((s[cur].islower() and s[cur+1].isupper()) or\
               (s[cur].isupper() and s[cur+1].islower())) and\
               (s[cur].lower()==s[cur+1].lower()):
                   s = s[:cur]+s[cur+2:]
                   cur -= 2
            
            cur += 1
        
        return s

File: 1544. Make The String Great.js
/**
 * @param {string} s
 * @return {string}
 */
 var makeGood = function(s) {
    let cnt = 0;

    while(cnt<s.length-1){
        if(cnt<0){
            cnt += 1;
            continue;
        }
        
        let [cur, next] = [s.charCodeAt(cnt), s.charCodeAt(cnt+1)];
        
        if (((cur>96 && cur<123 && next>64 && next<91) 
           ||(cur>64 && cur<91 && next>96 && next<123))
           &&(Math.abs(cur-next)==32)){

               s = s.slice(0, cnt)+s.slice(cnt+2);
               cnt-=2
        }
        cnt += 1
    }
    return s;
};